internal/ginutil/router.go

```diff
@@ -0,0 +1,34 @@
+package ginutil
+
+import (
+	"strings"
+
+	"github.com/gin-gonic/gin"
+)
+
+// TrailingSlashRouteHelper wraps a gin.IRouter and ensures every route registered with GET
+// has a corresponding alternate route with or without, dependent on the endpoint being registered,
+// a trailing slash.
+type TrailingSlashRouteHelper struct {
+	gin.IRouter
+}
+
+// GET overrides the internal gin.IRouter GET. It interrogates endpoint for a trailing slash. If
+// no trailing slash is present, it registers the endpoint and a corresponding endpoint with a
+// trailing slash using the same handler. If it does end in a trailing slash, it does the inverse.
+func (r TrailingSlashRouteHelper) GET(endpoint string, handler ...gin.HandlerFunc) gin.IRoutes {
+	// Determine if the alternate endpoint should end with a slash or have it stripped.
+	// This ensures we don't have routes such as
+	// 		/2009-04-04/meta-data/instance-id/
```

Comment:
`/2009-04-04/meta-data/instance-id/`

I believe trailing slash is only supported on array and hash tree items. I do not believe a trailing slash should be accepted with scalar values like strings, ints, and bools.

Can you confirm this behavior in EC2?

Reply:
I was suspecting that on EC2, `GET 169.254.254./2009-04-04/meta-data/instance-id/` would yield a 404.

However, after running against a mocking environment (perhaps not the same, but as close as I can get) this does not appear to be the case.

```
$ nerdctl run -it --rm -p 1338:1338 public.ecr.aws/aws-ec2/amazon-ec2-metadata-mock:v1.11.2
$ curl http://localhost:1338/latest/meta-data/instance-id
i-1234567890abcdef0
$ curl http://localhost:1338/latest/meta-data/instance-id/
i-1234567890abcdef0
```

https://github.com/aws/amazon-ec2-metadata-mock is Apache2 and may have other hints (or types and interfaces).

Reply:
Either way, the spec doesn't indicate anything so we're very much left to figure it out. I'm being guided by the things we integrate with - hopefully they never contradict.